src/reconcile/rec_write.c

```diff
@@ -344,6 +327,15 @@ static int  __rec_dictionary_init(WT_SESSION_IMPL *, WT_RECONCILE *, u_int);
 static int  __rec_dictionary_lookup(
 		WT_SESSION_IMPL *, WT_RECONCILE *, WT_KV *, WT_DICTIONARY **);
 static void __rec_dictionary_reset(WT_RECONCILE *);
+static inline int __rec_split_crossing_bnd(
+    WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t next_len);
+
+#define	CROSSING_ALT_BND(r, next_len)			\
+	(r->bnd[r->bnd_next].alt_offset == 0 && next_len > r->alt_space_avail)
+#define	CROSSING_SPLIT_BND(r, next_len)			\
+	(next_len > r->space_avail)
+#define	NEED_KEY_PROMOTE(r, next_len)			\
+	(CROSSING_ALT_BND(r, next_len) || CROSSING_SPLIT_BND(r, next_len))
```

Comment:
@sulabhM, you'll want to change these macro names, they step on name spaces outside of ours.